packages/modules/cockroachdb/src/cockroachdb-container.ts

```diff
@@ -0,0 +1,71 @@
+import { AbstractStartedContainer, GenericContainer, StartedTestContainer, Wait } from "testcontainers";
+
+const COCKROACH_PORT = 26257;
+
+export class CockroachDbContainer extends GenericContainer {
+  private database = "test";
+  private username = "test";
+
+  constructor(image = "cockroachdb/cockroach:v24.3.5") {
+    super(image);
+    this.withExposedPorts(COCKROACH_PORT)
+      .withCommand(["start-single-node", "--insecure"])
+      .withWaitStrategy(Wait.forLogMessage(/.*end running init files.*/, 1))
+      .withStartupTimeout(120_000);
+  }
+
+  public withDatabase(database: string): this {
+    this.database = database;
+    return this;
+  }
+
+  public withUsername(username: string): this {
+    this.username = username;
+    return this;
+  }
+
+  public override async start(): Promise<StartedCockroachDbContainer> {
+    this.withEnvironment({
+      COCKROACH_DATABASE: this.database,
+      COCKROACH_USER: this.username,
+    });
+    return new StartedCockroachDbContainer(await super.start(), this.database, this.username);
+  }
+}
+
+export class StartedCockroachDbContainer extends AbstractStartedContainer {
+  private readonly port: number;
+
+  constructor(
+    startedTestContainer: StartedTestContainer,
+    private readonly database: string,
+    private readonly username: string
+  ) {
+    super(startedTestContainer);
+    this.port = startedTestContainer.getMappedPort(COCKROACH_PORT);
```

Comment:
This needs to be calculated every time `getPort()` is called because the container can be restarted and get given a different port. E.g:

```js
const container = await new CockroachDbContainer().start();
const port = container.getFirstMappedPort();
await container.restart();
port !== container.getFirstMappedPort()
```

Reply:
Also, worth pointing out that currently this is not working for the postgres container. I'll make a follow up PR to fix it.